doc/source/dev/gitwash/development_workflow.rst

```diff
@@ -43,36 +43,16 @@ as clear as possible.
 Making a new feature branch
 ===========================
 
-First, update your master branch with changes that have been made in the main
-Numpy repository. In this case, the ``--ff-only`` flag ensures that a new
-commit is not created when you merge the upstream and master branches. It is
-very important to avoid merging adding new commits to ``master``.
+First, fetch new commits from the ``upstream`` repository:
 
 ::
 
-   # go to the master branch
-   git checkout master
-   # download changes from github
    git fetch upstream
-   # update the master branch
-   git merge upstream/master --ff-only
-   # Push new commits to your Github repo
-   git push
-
-.. note::
 
-   You could also use ``pull``, which combines ``fetch`` and ``merge``, as
-   follows::
-   
-        git pull --ff-only upstream master
+Then, create a new branch based on the master branch of the upstream
+repository::
 
-   However, never use ``git pull`` without explicity indicating the source
-   branch (as above); the inherent ambiguity can cause problems. This avoids a
-   common mistake if you are new to Git. 
-
-Finally create a new branch for your work and check it out::
-
-   git checkout -b my-new-feature master
+   git checkout -b my-new-feature upstream/master
```

Comment:
If you look at your `.git/config`, you have something like this there:
```
[remote "upstream"]
        url = git://github.com/scipy/scipy.git
        fetch = +refs/heads/*:refs/remotes/upstream/*
```
which determines what is fetched, and how the names are mapped.
Note that github stores pull requests in namespace `refs/pull/`, so if you're feeling brave, you can add a second fetch mapping:
```
[remote "upstream"]
        url = git://github.com/scipy/scipy.git
        fetch = +refs/heads/*:refs/remotes/upstream/*
        fetch = +refs/pull/*/head:refs/remotes/upstream/pr/*
```
so that after `git fetch` all pull requests appear as mirrored branches named along the lines of `upstream/pr/7300`.